stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/domain/channel/internal/ChannelDao.kt

```diff
@@ -36,11 +37,27 @@ internal interface ChannelDao {
     @Query("SELECT cid FROM stream_chat_channel_state")
     suspend fun selectAllCids(): List<String>
 
+    @Query(
+        "SELECT cid FROM stream_chat_channel_state " +
+            "WHERE syncStatus = :syncStatus " +
+            "ORDER BY syncStatus ASC " +
+            "LIMIT :limit"
+    )
+    suspend fun selectCidsSyncNeeded(
```

Comment:
Can we change to `selectCidsBySyncStatus`? A user could actually request another sync state using the syncStatus parameter. Same goes for the repository